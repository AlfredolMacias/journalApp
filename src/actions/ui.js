import { types } from "../types/types";

export const setError = ( err ) => ({
    type : types.uiSetError,
    payload: err
});

export const RemoveError = ( ) => ({
    type : types.uiRemoveError,

});

export const uiStartLoading = () => ({
    type: types.uiStartLoading
});

export const uiFinishLoading = () => ({
    type: types.uiFinishLoading,
});