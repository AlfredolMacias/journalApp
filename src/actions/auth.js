import { types } from "../types/types"
import { firebase, googleAuthProvider } from '../firebase/firebase-config';
import { uiFinishLoading, uiStartLoading } from "./ui";
import Swal from 'sweetalert2';
import { cleanNotes } from "./notes";

export const startLoginEmailPassword = ( email, password ) =>{
    return ( dispatch ) => {
        dispatch( uiStartLoading() );
        firebase.auth().signInWithEmailAndPassword(email, password)
            .then( ({user}) => {
                console.log("HOALAASs");
                dispatch( login (user.uid, user.displayName));
                dispatch( uiFinishLoading() );
            })
            .catch( e =>{
                dispatch( uiFinishLoading() );
                Swal.fire('Error', e.message, 'error');
            })
    }
}

export const startRegisterWithEmailPassword = ( name, email, password) => {
    return ( dispatch ) => {

        firebase.auth().createUserWithEmailAndPassword( email, password)
            .then( async ({ user }) => {
                
                await user.updateProfile({ displayName: name })
                dispatch( 
                    login(user.uid, user.displayName)
                )
            } ).catch( e =>{
                Swal.fire('Error', e.message, 'error');
            })

    }
}

export const startGoogleLogin = () => {
    return ( dispatch ) => {
        firebase.auth().signInWithPopup( googleAuthProvider )
            .then(( { user } )=> {
                dispatch(
                    login( user.uid, user.displayName)
                )
            })
    }
}

export const login = (uid, displayName) => ({
    type: types.login,
    payload:{
        uid,
        displayName
    }
})


export const startLogout = () => {
    return async ( dispatch ) => {
        await firebase.auth().signOut();
        dispatch( logout() );
        dispatch ( cleanNotes() );
    }
}

export const logout = () => ({
    type: types.logout
});

