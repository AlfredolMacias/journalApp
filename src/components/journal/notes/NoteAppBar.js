import React from 'react';
import moment from "moment";
import { useDispatch, useSelector } from 'react-redux';
import { startSaveNote, startUploading } from '../../../actions/notes';

export const NoteAppBar = () => {
    const dispatch = useDispatch();
    const {active} = useSelector(state => state.notes)
    const handleSave = () => {
        dispatch( startSaveNote( active ) );

    }
    const noteDate = moment(active.date);

    const handlePictureClick = () => {
        document.querySelector('#fileSelector').click();

    }

    const handlePictureChange = (e) => {
        const file = e.target.files[0];
        if( file ){
            dispatch( startUploading( file ) );
        }

    }

    return (
        <div className="notes__appbar">
            <span>{noteDate.format('LLL')}</span>
            <input 
                id = 'fileSelector'
                type = 'file'
                name = 'file'
                style = {{display: 'none'}}
                onChange = { handlePictureChange }    
            />    

            <div>
                <button className="btn" onClick = { handlePictureClick }>Picture</button>
                <button className="btn" onClick= { handleSave }>Save</button>
            </div>
        </div>
    )
}
