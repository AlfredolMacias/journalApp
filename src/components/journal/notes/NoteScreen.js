import React, { useEffect, useRef } from 'react';
import { useDispatch, useSelector } from "react-redux";
import { activeNote, startDeleting } from '../../../actions/notes';
import { useForm } from '../../../hooks/useForm';
import { NoteAppBar } from './NoteAppBar';

export const NoteScreen = () => {

    const { active:note } = useSelector(state => state.notes)
    const dispatch = useDispatch();
    const [ formValues, handleInputChange, reset ] = useForm( note );

    const { title, body, id } = formValues; 

    const activeId = useRef( note.id );

    useEffect(() => {
        if( note.id !== activeId.current ){
            reset( note );
            activeId.current = note.id
        }
    }, [note, reset])

    useEffect(() => {

        dispatch( activeNote(formValues.id, {...formValues}));
    }, [formValues, dispatch]);

    const handleDelete = () => {
        dispatch( startDeleting ( id ) );
    }

    return (
        <div className="notes__main-content">
            <NoteAppBar />

            <div className="notes__content">

                <input 
                    type='text'
                    placeholder="Something Cool..."
                    name = 'title'
                    className='notes__title-input'
                    value = { title }
                    autoComplete= 'off'
                    onChange = { handleInputChange }
                />

                <textarea
                    placeholder="What's up today?"
                    className="notes__textarea"
                    name = 'body'
                    value = { body }
                    autoComplete = 'off'
                    onChange = { handleInputChange }
                >
                </textarea>
                {
                    (note.url ) &&
                    <div className="notes__image">
                        <img alt="Algo" src={ note.url} />


                    </div>
                }

            </div>
            
            <button
                className="btn btn-danger"
                onClick = { handleDelete }
            >
                Eliminar
            </button>
        </div>
    )
}
