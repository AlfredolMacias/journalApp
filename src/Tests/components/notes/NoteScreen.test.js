import React from 'react';
import { mount } from "enzyme";
import { Provider } from "react-redux";

import configureStore from 'redux-mock-store';
import thunk from "redux-thunk";
import { activeNote } from '../../../actions/notes';
import { NoteScreen } from '../../../components/journal/notes/NoteScreen';


jest.mock('../../../actions/notes', () => ({
    activeNote: jest.fn(),
}))

const middlewares = [ thunk ]
const mockStore = configureStore(middlewares);

const initState = {
    auth:{
        uid: '1',
        name: 'Alfredo',

    },
    ui: {
        loading: false,
        msgError: null
    },
    notes:{
        active:{
            id: 1234,
            title:'Hola',
            body: 'Mundo',
            date: 0
        },
        notes:[]
    }
}

let  store = mockStore( initState )
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store = {store } >
        <NoteScreen />
    </Provider>
);

describe('Pruebas en el componente <NoteScreen/>', () => {

    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de disparar el activeNote', () => {

        wrapper.find('input[name="title"]').simulate('change',{
            target:{
                name: 'title',
                value: 'Hola Mundo'
            }
        });
        expect(activeNote).toHaveBeenLastCalledWith(
            1234,
            {
                title: 'Hola Mundo',
                body: 'Mundo',
                id: 1234,
                date:0
            }
        );
    })
    
    
    
})
