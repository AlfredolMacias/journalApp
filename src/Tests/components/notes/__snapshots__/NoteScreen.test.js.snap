// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <NoteScreen/> debe de mostrarse correctamente 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      December 31, 1969 6:00 PM
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Something Cool..."
      type="text"
      value="Hola"
    />
    <textarea
      autoComplete="off"
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="What's up today?"
      value="Mundo"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Eliminar
  </button>
</div>
`;
